### PR TITLE
Add ustream content provider - removed 2nd request

### PR DESCRIPTION
**Pull request by sethwebster**

### CODE REVIEW

File: JabbR/ContentProviders/UStreamContentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                Content = iframeHtml,
+                Title = response.ResponseUri.AbsoluteUri.ToString()
+            };
+        }
+
+        private string ExtractIFrameCode(HttpWebResponse response)
+        {
+            var iframeStr = new System.IO.StreamReader(response.GetResponseStream()).ReadToEnd();
+
+            var matches = _extractEmbedCodeRegex.Match(iframeStr)
```
Comment (davidfowl) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: You still need this duplicated thing?

File: JabbR/ContentProviders/UStreamContentProvider.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            return new ContentProviderResultModel()
+            {
+                Content = iframeHtml,
+                Title = response.ResponseUri.AbsoluteUri.ToString()
+            };
+        }
+
+        private string ExtractIFrameCode(HttpWebResponse response)
+        {
+            var iframeStr = new System.IO.StreamReader(response.GetResponseStream()).ReadToEnd();
```
Comment (davidfowl) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: Why is the full type name here? 
Who handles closing the HttpWebResponse itself, the caller? 
Where is the stream being closed? 
Are there any other content providers that don't clean up after themselves?

File: JabbR/ContentProviders/UStreamContentProvider.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+            var matches = _extractEmbedCodeRegex.Match(iframeStr)
+                                .Groups
+                                .Cast<Group>()
+                                .Skip(1)
+                                .Select(g => g.Value)
+                                .Where(v => !String.IsNullOrEmpty(v));
+
+            return matches.Count() > 0 ? matches.First() : string.Empty;
```
Comment (davidfowl) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: Uppercase String.Empty.
Why not use `matches.FirstOrDefault() ?? String.Empty`

File: JabbR/ContentProviders/UStreamContentProvider.cs (hunk 1 above)
Comment (sethwebster) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: This is a special case. ExtractParameters deals with the actual params in the URI; this is pulling it from the content. 

WDYT?

File: JabbR/ContentProviders/UStreamContentProvider.cs (hunk 2 above)
Comment (sethwebster) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: Theraflu. That's all I can say.

File: JabbR/ContentProviders/UStreamContentProvider.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Web;
+using JabbR.ContentProviders.Core;
+using System.IO;
```
Comment (davidfowl) [inline JabbR/ContentProviders/UStreamContentProvider.cs]: Sort usings.

**Pull request merged**